### PR TITLE
[Wax] Stop invalid Coinbase transactions from showing up in pending-transactions

### PR DESCRIPTION
**Pull request by WhoSoup**
This PR is a fix for #1006. This works by adding a flag to FBlocks that track whether or not the coinbase tx has been patched, and in return GetPendingTransactions only returns the coinbase if it has been patched. There are better ways to fix this, such as not creating the coinbase at all until the real timestamp is in, but the code frequently relies on the coinbase transaction existing. I figured this was the least-invasive method that doesn't affect existing behavior.

Changes:
* Add coinbasePatched flag to FBlock and support methods in both the block and the interface
* Add DBHeight to PendingTransaction result that indicates the expected height of the transaction (or 0 for transactions in holding)
* Change the consensus code to use the new FBlock functions instead of blind casts
* Change state.GetPendingTransactions to use string as parameter instead of interface
* Rewrite the GetPendingTransactions to remove completely unnecessary loop over VMs
* Fix console sim input that tried to send just a byte to state.GetPendingTransactions

### CODE REVIEW

**Comment by WhoSoup** [comment]
Rebased this to the wax_rollup branch.

File: state/stateConsensus.go [hunk 1]
```diff
@@ -2389,15 +2388,16 @@ func (s *State) ProcessDBSig(dbheight uint32, msg interfaces.IMsg) bool {
 			s.LogPrintf("dbsig", "1st ProcessDBSig(): %10s DBSig dbht %d leaderheight %d VMIndex %d Timestamp %x %d, leadertimestamp = %x %d",
 				s.FactomNodeName, dbs.DBHeight, s.LLeaderHeight, dbs.VMIndex, dbs.GetTimestamp().GetTimeMilli(), dbs.GetTimestamp().GetTimeMilli(), s.LeaderTimestamp.GetTimeMilliUInt64(), s.LeaderTimestamp.GetTimeMilliUInt64())
 
-			cbtx := fs.GetCurrentBlock().(*factoid.FBlock).Transactions[0].(*factoid.Transaction)
-			foo := cbtx.MilliTimestamp
+			cb := fs.GetCurrentBlock()
+			cbtx := cb.GetTransactions()[0]
+			foo := cbtx.GetTimestamp().GetTimeMilliUInt64()
```
Comment (PaulBernier) [inline state/stateConsensus.go:2393]: is this variable really named foo..... I know it's not you originally but from now on git will attribute it to you, so you may want to rename :P

Comment (WhoSoup) [reply state/stateConsensus.go]: > is this variable really named foo..... I know it's not you originally but from now on git will attribute it to you, so you may want to rename :P

Gave them all more appropriate names.

**Comment by WhoSoup** [comment]
> Just to be sure you have tested that change manually by calling the API?

Yeah, @ilzheev and I spent a couple hours working on this problem and fixing it.

**Pull request merged**